unbuffered.go

```diff
@@ -0,0 +1,116 @@
+package fasthttp
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+)
+
+type UnbufferedWriter interface {
+	Write(p []byte) (int, error)
+	WriteHeaders() (int, error)
+	Close() error
+}
+
+type UnbufferedWriterHttp1 struct {
```

Comment:
I don't think this type should be exposed, it can be kept internally. I also wouldn't add `Http1` to the name as everything in fastthttp is http1.